python/mxnet/gluon/estimator/estimator.py

```diff
@@ -175,14 +173,45 @@ def _batch_fn(self, batch, ctx, is_iterator=False):
         label = gluon.utils.split_and_load(label, ctx_list=ctx, batch_axis=0)
         return data, label
 
+    def infer_data_info(self, data):
+        """Retrieve the data information such as batch size,
+        Number of batches, and total number of samples
+
+        Parameters
+        ----------
+        data : DataLoader
+            A DataLoader instance with data and/or label
+
+        Returns
+        -------
+        num_batches: int
+            Number of batches the data is divided into
+        total_samples: int
+            Total Number of samples
+        batch_size: int
+            Batch size
+        """
+        if isinstance(data, gluon.data.DataLoader):
+            num_batches = len(data)
+            total_samples = len(data._dataset)
```

Comment:
this could be an expensive operation.

Reply:
If the DataLoader created using `MXNet.gluon.data.dataset.ArrayDataset` then the `ArrayDataset` has the instance variable `_length` which infers the length of an actual data. Another way to infer the total number of sample is to extract the `shape` information from the data.